packages/module/src/Hooks/pagination.test.tsx

```diff
@@ -49,9 +49,81 @@ describe('useDataViewPagination', () => {
     expect(result.current).toEqual({ 
       onPerPageSelect: expect.any(Function),
       onSetPage: expect.any(Function),
-      page: 3,
+      page: 1,
       perPage: 50
     })
   });
 
+  it('should read pagination state from URL', () => {
+    const mockSearchParams = new URLSearchParams('page=2&perPage=10');
+    const { result } = renderHook(() =>
+      useDataViewPagination({
+        searchParams: mockSearchParams,
+        setSearchParams: jest.fn(),
+        page: 1,
+        perPage: 5,
+      })
+    );
+
+    expect(result.current).toEqual({
+      onPerPageSelect: expect.any(Function),
+      onSetPage: expect.any(Function),
+      page: 2,
+      perPage: 10,
+    });
+  });
+
+  it('should set pagination state in URL when page changes', () => {
+    const mockSetSearchParams = jest.fn();
+    const { result } = renderHook(() =>
+      useDataViewPagination({
+        searchParams: new URLSearchParams(),
+        setSearchParams: mockSetSearchParams,
+        page: 1,
+        perPage: 5,
+      })
+    );
+
+    act(() => {
+      result.current.onSetPage(undefined, 4);
+    });
+
+    expect(mockSetSearchParams).toHaveBeenCalledTimes(2);
```

Comment:
You check if the setter was called but not what values were passed to the argument.

Reply:
Oh yeah I forgot to finish this one